Marlin/src/gcode/config/M43.cpp

```diff
@@ -143,68 +147,94 @@ inline void servo_probe_test() {
+        SERIAL_ECHOLNPGM("= Mechanical Switch detected");
+        if (deploy_state) {
+          SERIAL_ECHOLNPGM("  DEPLOYED state: HIGH (logic 1)");
+          SERIAL_ECHOLNPGM("  STOWED (triggered) state: LOW (logic 0)");
+        }
+        else {
+          SERIAL_ECHOLNPGM("  DEPLOYED state: LOW (logic 0)");
+          SERIAL_ECHOLNPGM("  STOWED (triggered) state: HIGH (logic 1)");
+        }
+        #if ENABLED(BLTOUCH)
+          SERIAL_ECHOLNPGM("ERROR: BLTOUCH enabled - set this device up as a Z Servo Probe with inverting as true.");
+        #endif
+        return;
+      }
+    }
 
-        if (deploy_state != READ(PROBE_TEST_PIN)) { // probe triggered
+    // Ask the user for a trigger event and measure the pulse width
+    // Since it could be a real servo or a BLTouch (any kind) or clone
+    // use only "common" functions - i.e. SERVO_MOVE. No bltouch.xxxx stuff.
+    MOVE_SERVO(probe_index, servo_angles[Z_PROBE_SERVO_NR][0]); // Deploy
+    safe_delay(500);
+    SERIAL_ECHOLNPGM("** Please trigger probe within 30 sec **");
+    uint16_t probe_counter = 0;
 
-          for (probe_counter = 1; probe_counter < 50 && deploy_state != READ(PROBE_TEST_PIN); ++probe_counter)
-            safe_delay(2);
+    // Allow 30 seconds max for operator to trigger probe
+    for (uint16_t j = 0; j < 500 * 30 && probe_counter == 0 ; j++) {
+      safe_delay(2);
 
-          if (probe_counter == 50)
-            SERIAL_ECHOLNPGM("Z Servo Probe detected"); // >= 100mS active time
-          else if (probe_counter >= 2)
-            SERIAL_ECHOLNPAIR("BLTouch compatible probe detected - pulse width (+/- 4mS): ", probe_counter * 2); // allow 4 - 100mS pulse
-          else
-            SERIAL_ECHOLNPGM("noise detected - please re-run test"); // less than 2mS pulse
+      if (0 == j % (500 * 1)) gcode.reset_stepper_timeout();    // Keep steppers powered
 
-          MOVE_SERVO(probe_index, servo_angles[Z_PROBE_SERVO_NR][1]); // Stow
+      if (deploy_state != READ(PROBE_TEST_PIN)) {               // probe triggered
+        for (probe_counter = 1; probe_counter < 15 && deploy_state != READ(PROBE_TEST_PIN); ++probe_counter) safe_delay(2);
 
-        }  // pulse detected
+        SERIAL_ECHOLNPAIR(". Pulse width (+/- 4mS): ", probe_counter * 2);
 
-      } // for loop waiting for trigger
+        if (blt && probe_counter == 15) SERIAL_ECHOLNPGM("= BLTouch V3.1 detected");
+        else if (probe_counter == 15)   SERIAL_ECHOLNPGM("= Z Servo Probe detected");
+        else if (probe_counter >= 4)    SERIAL_ECHOLNPGM("= BLTouch pre V3.1 or compatible probe detected");
+        else                            SERIAL_ECHOLNPGM("ERROR: noise detected - please re-run test");
+        MOVE_SERVO(probe_index, servo_angles[Z_PROBE_SERVO_NR][1]); // Stow
+        return;
+      }
+    }
 
-      if (probe_counter == 0) SERIAL_ECHOLNPGM("trigger not detected");
+    if (probe_counter == 0) SERIAL_ECHOLNPGM("ERROR: Trigger not detected");
```

Comment:
Remember that this will cause OctoPrint to disconnect by default.

Reply:
@thinkyhead That's a good point, but Gina has made OctoPint be little more discerning nowadays. I tested this with a bogus "ERROR:" message, see below. Still she laments rightly the lack of standardisation for fatal error messages, therefore I agree with you totally that "FAIL:" is better
![grafik](https://user-images.githubusercontent.com/5990805/57786418-baa62980-7733-11e9-9b48-4476aa4bb926.png)